### PR TITLE
Add pre-commit commands

### PR DESCRIPTION
**Pull request by jirikuncar**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue, or add one feature, at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md).

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,16 +1,49 @@
...
+  stages: [manual]
+- id: push
+  name: Run apigentools push
+  language: python
+  entry: apigentools push
+  description: Run apigentools push inside this repository.
+  pass_filenames: false
+  always_run: true
+  stages: [manual]
+- id: apigentools
```
Comment (nmuesch) [inline .pre-commit-hooks.yaml:39]: whats this one for? Just debugging?

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -1,16 +1,49 @@
...
-      - validate
-    description: Validate OpenAPI specification files generated by apigentools using configuration inside this repository using container wrapper.
-    pass_filenames: false
+- id: validate
+  name: Validate OpenAPI specification files
+  language: python
+  entry: apigentools validate
+  description: Validate OpenAPI specification files generated by apigentools using configuration inside this repository.
+  pass_filenames: false
+- id: templates
```
Comment (nmuesch) [inline .pre-commit-hooks.yaml:7]: Do we explicitly need templates if its inside generate?

Comment (jirikuncar) [reply .pre-commit-hooks.yaml:7]: No, but @bkabrda wants separate steps in CI.

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (jirikuncar) [reply .pre-commit-hooks.yaml:39]: yes. and also one can define custom command easily using `args`

**Pull request merged**